rust-symcrypt/src/hmac.rs

```diff
@@ -0,0 +1,215 @@
+use core::ffi::c_void;
+use std::mem;
+use std::ptr;
+use symcrypt_sys;
+
+pub const SHA256_HMAC_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA256_RESULT_SIZE as usize;
+pub const SHA384_HMAC_RESULT_SIZE: usize = symcrypt_sys::SYMCRYPT_SHA384_RESULT_SIZE as usize;
+
+pub struct HmacSha256State {
+    state: symcrypt_sys::SYMCRYPT_HMAC_SHA256_STATE,
+    expanded_key: symcrypt_sys::SYMCRYPT_HMAC_SHA256_EXPANDED_KEY,
+}
+
+impl HmacSha256State {
+    pub fn new(key: &[u8]) -> Self {
+        let mut instance = HmacSha256State {
+            state: symcrypt_sys::SYMCRYPT_HMAC_SHA256_STATE::default(),
+            expanded_key: symcrypt_sys::SYMCRYPT_HMAC_SHA256_EXPANDED_KEY::default(),
+        };
+        unsafe {
+            symcrypt_sys::SymCryptHmacSha256ExpandKey(
+                &mut instance.expanded_key,        // pExpandedKey
+                key.as_ptr(),                      // pbKey
+                key.len() as symcrypt_sys::SIZE_T, // cbKey
+            );
+            symcrypt_sys::SymCryptHmacSha256Init(&mut instance.state, &mut instance.expanded_key);
```

Comment:
>mut [](http://example.com/codeflow?start=71&length=3)

Does this need to be mut? Its an in read only parameter.

Reply:
Changed to non mut for expanded_key